src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineViewFactory.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Tagging;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Editor.Tagging;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    /// <summary>
+    /// Responsible for wiring up all the different components of the document outline feature.
+    /// </summary>
+    internal static class DocumentOutlineViewFactory
+    {
+        public static DocumentOutlineView CreateView(
+            ILanguageServiceBroker2 languageServiceBroker,
+            IThreadingContext threadingContext,
+            IAsynchronousOperationListener asyncListener,
+            IVsEditorAdaptersFactoryService editorAdaptersFactoryService,
+            IVsCodeWindow codeWindow)
+        {
+            threadingContext.ThrowIfNotOnUIThread();
+            var (textViewEventSource, textBuffer) = CreateEventSource(asyncListener, editorAdaptersFactoryService, codeWindow);
+            var viewModel = new DocumentOutlineViewModel(languageServiceBroker, asyncListener, textViewEventSource, textBuffer, threadingContext);
+            return new DocumentOutlineView(viewModel, editorAdaptersFactoryService, codeWindow);
+        }
+
+        private static (CompilationAvailableTaggerEventSource, ITextBuffer) CreateEventSource(
+            IAsynchronousOperationListener asyncListener,
+            IVsEditorAdaptersFactoryService editorAdaptersFactoryService,
+            IVsCodeWindow codeWindow)
+        {
+            if (ErrorHandler.Failed(codeWindow.GetLastActiveView(out var textView)))
+            {
+                FailFast.Fail("Unable to get the last active text view. IVsCodeWindow implementation we are given is invalid.");
```

Comment:
uh... no :)   this tears down VS entirely.  let's not do that.  if we can't make this, then we should gracefully bail out.

Reply:
The VS editor implementation is not valid. This would mean that the VS editor has constructed us but has not active windows which breaks the API. I do not know what graceful recovery from this would be. I suppose we just show a blank window but the editor is in an invalid state.